### PR TITLE
Almost full ROS 1 parity

### PR DESCRIPTION
**Pull request by rjoomen**
- tesseract_rviz works
- Most examples work
- Plugins cleaned up and working
- tesseract_monitoring works, except contact monitor
- planning server does not work yet
- CI for ROS2 Humble on Ubuntu 22.04 (Jammy) and for ROS2 Foxy on Ubuntu 20.04 (Focal) work

### CODE REVIEW

File: tesseract_rviz/src/ros_manipulation_widget.cpp [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #include <tesseract_rviz/ros_manipulation_widget.h>
 #include <tesseract_rviz/conversions.h>
+#include <tesseract_rviz/conversions.h>
+#include <tesseract_rviz/conversions.h>
```
Comment (marrts) [inline tesseract_rviz/src/ros_manipulation_widget.cpp:4]: duplicate includes

**Review by marrts — COMMENTED** [review]
Thank you so much for putting this all together, been wanting to clean things up for a while. It worked for me just fine in Foxy when I pulled it down. Just a couple questions and a couple suggestions.

File: tesseract_monitoring/include/tesseract_monitoring/environment_monitor.h [hunk 2]
```diff
@@ -196,28 +197,28 @@ class ROSEnvironmentMonitor : public tesseract_environment::EnvironmentMonitor
...
+  // called by state_update_timer_ when a state update is pending
   void updateJointStateTimerCallback();
 
   // Callback for a new state msg
-  void newEnvironmentStateCallback(const tesseract_msgs::msg::EnvironmentState::SharedPtr env);
+  void newEnvironmentStateCallback(const tesseract_msgs::msg::EnvironmentState::ConstSharedPtr env);
 
   /** @brief Callback for modifying the environment via service request */
-  void modifyEnvironmentCallback(tesseract_msgs::srv::ModifyEnvironment::Request::SharedPtr req,
+  bool modifyEnvironmentCallback(tesseract_msgs::srv::ModifyEnvironment::Request::SharedPtr req,
```
[the remark this answers is not in the document]
Comment (rjoomen) [reply tesseract_monitoring/include/tesseract_monitoring/environment_monitor.h:209]: That's directly taken from tesseract_ros1. These callbacks all return [a success state](http://wiki.ros.org/roscpp/Overview/Services#Callback_Signature). Upon further investigation, it seems this is not supported in ROS2, though, [see here](https://answers.ros.org/question/356549/how-to-handle-exceptions-in-the-user-defined-callback-in-service-server-and-how-to-inform-service-client-about-the-failed-service-call-in-ros2-rclcpp/) for example.

Any ideas how to solve this and have the services return success/failure in ROS2? For now, returning a bool doesn't seem to be a problem, so we could keep it to make the code match tesseract_ros1 as much as possible.

[the remark this answers is not in the document]
Comment (marrts) [reply tesseract_monitoring/include/tesseract_monitoring/environment_monitor.h:209]: I would prefer to match the ROS2 standard which is returning a void. The success/failure of a request should be determined in the response of the service, not in the true/false of the callback. I know this deviates from tesseract_ros1, but it'll be more in line with ROS2.

**Pull request merged**